metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1611,63 +1644,82 @@ def emit_published_datasources(self) -> Iterable[MetadataWorkUnit]:
-        ) in self.upstream_tables.items():
-            if not is_embedded and not self.config.ingest_tables_external:
-                logger.debug(
-                    f"Skipping external table {table_urn} as ingest_tables_external is set to False"
-                )
-                continue
+        database_table_id_to_urn_map: Dict[str, str] = dict()
+        for urn, tbl in self.database_tables.items():
+            database_table_id_to_urn_map[tbl.id] = urn
+        tables_filter = f"{tableau_constant.ID_WITH_IN}: {json.dumps(list(database_table_id_to_urn_map.keys()))}"
+
+        for table in self.get_connection_objects(
+            database_tables_graphql_query,
+            tableau_constant.DATABASE_TABLES_CONNECTION,
+            tables_filter,
+        ):
+            yield from self.emit_table(table, database_table_id_to_urn_map)
 
-            dataset_snapshot = DatasetSnapshot(
-                urn=table_urn,
-                aspects=[],
+    def emit_table(
+        self, table: dict, database_table_id_to_urn_map: Dict[str, str]
+    ) -> Iterable[MetadataWorkUnit]:
+        database_table = self.database_tables[
+            database_table_id_to_urn_map[table[tableau_constant.ID]]
+        ]
+        columns = table.get(tableau_constant.COLUMNS, [])
+        is_embedded = table.get(tableau_constant.IS_EMBEDDED) or False
+        if not is_embedded and not self.config.ingest_tables_external:
+            logger.debug(
+                f"Skipping external table {database_table.urn} as ingest_tables_external is set to False"
             )
-            if browse_path:
-                # Browse path
-                browse_paths = BrowsePathsClass(
-                    paths=[f"/{self.config.env.lower()}/{self.platform}/{browse_path}"]
-                )
-                dataset_snapshot.aspects.append(browse_paths)
-            else:
-                logger.debug(f"Browse path not set for table {table_urn}")
-            schema_metadata = None
-            if columns:
-                fields = []
-                for field in columns:
-                    if field.get(tableau_constant.NAME) is None:
-                        logger.warning(
-                            f"Skipping field {field[tableau_constant.ID]} from schema since its name is none"
-                        )
-                        continue
-                    nativeDataType = field.get(
-                        tableau_constant.REMOTE_TYPE, tableau_constant.UNKNOWN
-                    )
-                    TypeClass = FIELD_TYPE_MAPPING.get(nativeDataType, NullTypeClass)
+            return
 
-                    schema_field = SchemaField(
-                        fieldPath=field[tableau_constant.NAME],
-                        type=SchemaFieldDataType(type=TypeClass()),
-                        description="",
-                        nativeDataType=nativeDataType,
+        dataset_snapshot = DatasetSnapshot(
```

Comment:
We should get rid of DatasetSnapshot if possible in the long term

Reply:
Right. I just moved the existing pieces around, will track and address this in followup, if its okay.